lib/resty/core/shdict.lua

```diff
@@ -372,22 +370,39 @@ local function shdict_incr(zone, key, value, init)
                 return error("bad init arg: number expected, got " .. typ)
             end
         end
+    end
 
-        has_init = 1
+    if init_ttl ~= nil then
+        local typ = type(init_ttl)
+        if typ ~= "number" then
+            init_ttl = tonumber(init_ttl)
+
+            if not init_ttl then
+                error("bad init_ttl arg: number expected, got " .. typ, 2)
+            end
+        end
+
+        if init_ttl < 0 then
+            error('bad "init_ttl" argument', 2)
+        end
+
+        if not init then
+            error('must provide "init" when providing "init_ttl"', 2)
+        end
```

Comment:
I'm not really happy with such complex argument value check. It looks branchy and expensive. Will you simplify it in a future PR? Thanks!